src/main/java/impl/image/ConvolutionProviderFactory.java

```diff
@@ -2,8 +2,41 @@
 
 import api.image.ConvolutionProvider;
 
+import java.awt.*;
+import java.lang.Math.*;
 public class ConvolutionProviderFactory {
-    public static ConvolutionProvider getInstance() {
-        return null;
+    private static ConvolutionProviderFactory instance;
+    public static ConvolutionProviderFactory getInstance()
+    {
+        if (instance == null)instance = new ConvolutionProviderFactory();
```

Comment:
It is important to use `{` `}` for all expression types. 
https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html#431